### PR TITLE
Update the design of DLRover

### PR DESCRIPTION
**Pull request by workingloong**

### CODE REVIEW

File: docs/design/easydl.md [hunk 1]
```diff
@@ -1,21 +1,21 @@
-# EasyDL Design
```
Comment (nash635) [inline docs/design/easydl.md:1]: 文件名也改一下吧

File: docs/design/dlrover-overview.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
+# DLRover Design
```
Comment (nash635) [inline docs/design/dlrover-overview.md:1]: docs/figures/easydl-overview.jpg
这个文件名也改一下吧

File: docs/design/dlrover-overview.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+deep learning training jobs. Instead, DLRover can pick up the appropriate resource 
+configuration for each job smartly and continue to optimize those jobs during their runtime.
+
+Currently, DLRover has supported automatic resource configuration. 
+However, the final goal of DLRover is to make the whole deep learning model
+training automatic.
+
+## Background
+
+In parameter server based jobs, all relevant parameters of a model
```
Comment (nash635) [inline docs/design/dlrover-overview.md:15]: server based -> server-based

File: docs/design/dlrover-overview.md [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+
+## Background
+
+In parameter server based jobs, all relevant parameters of a model 
+are distributed across parameter servers. Each worker takes partial training data
+as input and compute gradients of parameters. After that, the worker node sends 
+update gradients to the parameter server node which is keeping parameters. 
+
+However, model developers (users) have to learn more rather than model training 
+algorithms when they are using those jobs to train their models. In order to
```
Comment (nash635) [inline docs/design/dlrover-overview.md:21]: In order to -> To

File: docs/design/dlrover-overview.md [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+
+In parameter server based jobs, all relevant parameters of a model 
+are distributed across parameter servers. Each worker takes partial training data
+as input and compute gradients of parameters. After that, the worker node sends 
+update gradients to the parameter server node which is keeping parameters. 
+
+However, model developers (users) have to learn more rather than model training 
+algorithms when they are using those jobs to train their models. In order to 
+run a training job, those users have to specify the required resources for their 
+this job. Then the kubernetes cluster can allocate the required resources and
```
Comment (nash635) [inline docs/design/dlrover-overview.md:23]: kubernetes -> Kubernetes

File: docs/design/dlrover-overview.md [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+In parameter server based jobs, all relevant parameters of a model 
+are distributed across parameter servers. Each worker takes partial training data
+as input and compute gradients of parameters. After that, the worker node sends 
+update gradients to the parameter server node which is keeping parameters. 
+
+However, model developers (users) have to learn more rather than model training 
+algorithms when they are using those jobs to train their models. In order to 
+run a training job, those users have to specify the required resources for their 
+this job. Then the kubernetes cluster can allocate the required resources and 
+start the job. Unfortunately, we found it is quite ineffective way to ask the users
```
Comment (nash635) [inline docs/design/dlrover-overview.md:24]: quite -> quite an

File: docs/design/dlrover-overview.md [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+to take care the resource configuration. At first, the users are usually the
```
Comment (nash635) [inline docs/design/dlrover-overview.md:25]: take care of

File: docs/design/dlrover-overview.md [hunk 8]
```diff
@@ -0,0 +1,180 @@
...
+as input and compute gradients of parameters. After that, the worker node sends 
+update gradients to the parameter server node which is keeping parameters. 
+
+However, model developers (users) have to learn more rather than model training 
+algorithms when they are using those jobs to train their models. In order to 
+run a training job, those users have to specify the required resources for their 
+this job. Then the kubernetes cluster can allocate the required resources and 
+start the job. Unfortunately, we found it is quite ineffective way to ask the users
+to take care the resource configuration. At first, the users are usually the 
+experts on model training but not training jobs and kubernetes cluster. It is
```
Comment (nash635) [inline docs/design/dlrover-overview.md:26]: kubernetes -> Kubernetes

File: docs/design/dlrover-overview.md [hunk 9]
```diff
@@ -0,0 +1,180 @@
...
+algorithms when they are using those jobs to train their models. In order to 
+run a training job, those users have to specify the required resources for their 
+this job. Then the kubernetes cluster can allocate the required resources and 
+start the job. Unfortunately, we found it is quite ineffective way to ask the users
+to take care the resource configuration. At first, the users are usually the 
+experts on model training but not training jobs and kubernetes cluster. It is 
+not an easy task for them to have the optimal configuration at the first place.
+Secondly, a training job's resources requirement may vary during its runtime.
+A static resource configuration usually can not be the optimal one all the time.
+We have observe jobs failure, poor performance and inefficient resources since
```
Comment (nash635) [inline docs/design/dlrover-overview.md:30]: observe -> observed

File: docs/design/dlrover-overview.md [hunk 10]
```diff
@@ -0,0 +1,180 @@
...
+the users fail to provide the optimal resource configuration to their jobs.
```
Comment (nash635) [inline docs/design/dlrover-overview.md:31]: to their jobs -> for their jobs

File: docs/design/dlrover-overview.md [hunk 11]
```diff
@@ -0,0 +1,180 @@
...
+(e.g., CPU, memory usage, and training progress)
+and reports them to EasyDL Brain to persist those data into database
+periodically. Based on the job’s running status, 
+EasyDL Brain picks up appropriate algorithms to
+generate new resource plans and informs Elastic Trainer to
+start resources adjustment.
+
+### ElasticJob
+ElasticJob is a customized k8s controller to support elastic scheduling
+of Pods for DL training jobs. ElasticJob is reponsible to launch/delete
```
Comment (nash635) [inline docs/design/dlrover-overview.md:69]: reponsible -> responsible

File: docs/design/dlrover-overview.md [hunk 12]
```diff
@@ -0,0 +1,180 @@
...
+Pods on a k8s cluster according to a Scale CRD.
+We can apply a Scale CRD to scale up/down the number of
+parameter servers or workers and relaunch new Pods with more resource
+to replace Pods with insuffient resource. For example, we relaunch
```
Comment (nash635) [inline docs/design/dlrover-overview.md:73]: insuffient resource -> insufficient resources

File: docs/design/dlrover-overview.md [hunk 13]
```diff
@@ -0,0 +1,180 @@
...
+Elastic Trainer mainly contains two components:
+
+1. the training master which collects runtime statistics of all Pods,
+generates a Scale CRD notify ElasticJob, and dispatches data shards
+to workers.
+2. the elastic agent which runs on each Pod of the job. The elastic
+agent samples resource workload (e.g., CPU and memory usage)
+of the Pod and reports the workload
+data to the training master. The elastic agent of the worker also
+queries data shards form the training master to build input data
```
Comment (nash635) [inline docs/design/dlrover-overview.md:100]: form -> from

File: docs/design/dlrover-overview.md [hunk 14]
```diff
@@ -0,0 +1,180 @@
...
+pipeline of training frameworks.
+
+#### Dynamic Data Sharding
+
+The training master split the dataset into shards and the Elastic Agent
+of the worker quries the shard to train a model. Note a shard does not
```
Comment (nash635) [inline docs/design/dlrover-overview.md:106]: quries -> queries

File: docs/design/dlrover-overview.md [hunk 15]
```diff
@@ -0,0 +1,180 @@
...
+contain samples directly. Instead, a shard only includes indices of
+those samples. All shards are placed into a TODO queue.
+After a worker starts to run, the data input pipeline of a worker
+will query one shard from Elastic Trainer and read samples by indices
+in the shard. Meanwhile, Data Shard Manager marks this shard with the
+id of the worker and moves the shard from the TODO to the DOING queue. 
+After a worker consumes samples in the shard and update the new parameters in PS,
```
Comment (nash635) [inline docs/design/dlrover-overview.md:113]: update -> updates

File: docs/design/dlrover-overview.md [hunk 16]
```diff
@@ -0,0 +1,208 @@
...
+is running.
+
+There are 3 scenarios to satisfy different requirements from the simple
+to the complex.
+
+### Elasticity of a Single Job 
+In the scenario, users can launch an elastic training job and manually scale
+up/down Pods of the job.
+
+### Auto-scaling of a single Job
```
Comment (samplise) [inline docs/design/dlrover-overview.md:50]: single ->Single

File: docs/design/dlrover-overview.md [hunk 17]
```diff
@@ -0,0 +1,208 @@
...
+configuration completely and focus on the model training itself. Without any
+input (on resource configuration), DLRover can still provide the optimal
+resource plan for each training job, Meanwhile, DLRover can optimize the 
+performance of training jobs further through resource adjustment when a job
+is running.
+
+There are 3 scenarios to satisfy different requirements from the simple
+to the complex.
+
+### Elasticity of a Single Job
```
Comment (samplise) [inline docs/design/dlrover-overview.md:46]: Manual Scaling of a Single Job

**Review by samplise — APPROVED** [review]

**Pull request merged**